rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,311 @@
+    globaldb_set_unique_cache_value,
+)
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import CacheType, ChainID, ChecksumEvmAddress, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.db.drivers.gevent import DBCursor
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class GearboxPoolData:
+    """Mapping of gearbox pool rewards address to underlying pool name"""
+    pool_address: ChecksumEvmAddress
+    pool_name: str
+    farming_pool_token: str
+    lp_tokens: set[str]
+    str_chain_id: str
+    underlying_token: ChecksumEvmAddress | None = None
+
+
+def get_existing_pools(
+        cursor: 'DBCursor',
+        cache_type: Literal[CacheType.GEARBOX_POOL_ADDRESS],
+        chain_id: str,
+) -> set[ChecksumEvmAddress]:
+    """Returns all the gearbox pool rewards address stored in cache"""
+    return {
+        string_to_evm_address(address) for address in globaldb_get_general_cache_values(
+            cursor=cursor,
+            key_parts=(cache_type, chain_id),
+        )
+    }
+
+
+def save_gearbox_data_to_cache(
+        write_cursor: 'DBCursor',
+        database: Optional['DBHandler'],  # pylint: disable=unused-argument
+        new_data: list[GearboxPoolData],
+) -> None:
+    """Stores data about gearbox pools and their names in the global db cache tables."""
+    for pool in new_data:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.GEARBOX_POOL_ADDRESS, pool.str_chain_id),
+            values=[pool.pool_address],
+        )
+        globaldb_set_unique_cache_value(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.GEARBOX_POOL_NAME, pool.pool_address, pool.str_chain_id),
+            value=pool.pool_name,
+        )
+        globaldb_set_unique_cache_value(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.GEARBOX_POOL_FARMING_TOKEN, pool.pool_address, pool.str_chain_id),
+            value=pool.farming_pool_token,
+        )
+        if pool.lp_tokens is not None:
+            for idx, lp_token in enumerate(pool.lp_tokens):
+                globaldb_set_general_cache_values(
+                    write_cursor=write_cursor,
+                    key_parts=(CacheType.GEARBOX_POOL_LP_TOKENS, pool.pool_address, pool.str_chain_id, str(idx)),  # noqa: E501
+                    values=[lp_token],
+                )
+
+
+def read_ethereum_gearbox_data_from_cache() -> tuple[dict[ChecksumEvmAddress, Any]]:
+    return read_gearbox_data_from_cache(str_chain_id=str(ChainID.ETHEREUM.value))
```

Comment:
sorry seems that github didn't publish the comment but this function is just a thin wrapper used once. Where this function is called you can do:

```
= lambda: read_gearbox_data_from_cache(str_chain_id=str(ChainID.ETHEREUM.value))
```

and avoid this one